### ISSUE TITLE
Polygon.cut:  TypeError: Cannot read properties of undefined (reading 'edge_after')

### ISSUE THREAD

**Issue by Fauny** [alexbol99/flatten-js#176]
```js
{
  let { Polygon, Multiline, segment } = Flatten
  let polygon = new Polygon([
  [
    { ps: { x: 310.1, y: 423.82, name: "point", },
      pe: { x: 310.1, y: 460.66, name: "point", },
      name: "segment", },
    { ps: { x: 310.1, y: 460.66, name: "point", },
      pe: { x: 189.98, y: 460.66, name: "point", },
      name: "segment", },
    { ps: { x: 189.98, y: 460.66, name: "point", },
      pe: { x: 189.98, y: 423.94, name: "point", },
      name: "segment", },
    { ps: { x: 189.98, y: 423.94, name: "point", },
      pe: { x: 210.5, y: 423.94, name: "point", },
      name: "segment", },
    { ps: { x: 210.5, y: 423.94, name: "point", },
      pe: { x: 210.5, y: 313.75, name: "point", },
      name: "segment", },
    { ps: { x: 210.5, y: 313.75, name: "point", },
      pe: { x: 272.66, y: 313.75, name: "point", },
      name: "segment", },
    { ps: { x: 272.66, y: 313.75, name: "point", },
      pe: { x: 272.66, y: 325.99, name: "point", },
      name: "segment", },
    { ps: { x: 272.66, y: 325.99, name: "point", },
      pe: { x: 291.38, y: 325.99, name: "point", },
      name: "segment", },
    { ps: { x: 291.38, y: 325.99, name: "point", },
      pe: { x: 291.38, y: 423.94, name: "point", },
      name: "segment", },
    { ps: { x: 291.38, y: 423.94, name: "point", },
      pe: { x: 310.1, y: 423.82, name: "point", },
      name: "segment", },
  ]])

  let ml = new Multiline([
    {
      ps: { x: 210.5, y: 460.66, name: "point" },
      pe: { x: 210.5, y: 423.94, name: "point" },
      name: "segment",
    }, { 
      ps: { x: 210.5, y: 423.94, name: "point", },
      pe: { x: 253.94, y: 423.94, name: "point", },
      name: "segment",
    },
    {
      ps: { x: 253.94, y: 423.94, name: "point", },
      pe: { x: 253.94, y: 442.3, name: "point", },
      name: "segment",
    },
    {
      ps: { x: 253.94, y: 442.3, name: "point", },
      pe: { x: 272.66, y: 442.3, name: "point", },
      name: "segment",
    },
    {
      ps: { x: 272.66, y: 442.3, name: "point", },
      pe: { x: 272.66, y: 460.66, name: "point", },
      name: "segment",
    },
  ].map(l=>segment(l)))

   let svg = polygon.svg()
    svg += ml.svg({stroke:'red'})
    let stage = d3.select(DOM.svg(1000, 600))
    stage.html(svg)
  
    let slices = polygon.cut(ml)
  
    if (slices.toArray().length != 3) {
      console.error('Cut fail.')
    }
    
    return stage.node();
}
```

**Comment by Fauny** [comment]
@alexbol99 another error.

**Comment by alexbol99** [comment]
Hello @Fauny ,
I released a new patch v1.5.4, which fixes an error.
But the result after cut is two faces, like in the image.
I think this is the best that I can do for now.

<img width="488" alt="image" src="https://github.com/alexbol99/flatten-js/assets/6965440/a9988931-3ae2-4ff8-8202-28f25c39c7d8">

**Comment by Fauny** [comment]
@alexbol99 Thanks !!!
